backend/src/Controller/Admin/StoreOwnerPayment/AdminStoreOwnerPaymentController.php

```diff
@@ -0,0 +1,198 @@
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use OpenApi\Annotations as OA;
+
+/**
+ * Create and fetch Payments from store.
+ * @Route("v1/admin/storeownerpayment/")
+ */
+class AdminStoreOwnerPaymentController extends BaseController
+{
+    private AutoMapping $autoMapping;
+    private ValidatorInterface $validator;
+    private AdminStoreOwnerPaymentService $adminStoreOwnerPaymentService;
+
+    public function __construct(SerializerInterface $serializer, AutoMapping $autoMapping, ValidatorInterface $validator, AdminStoreOwnerPaymentService $adminStoreOwnerPaymentService)
+    {
+        parent::__construct($serializer);
+        $this->autoMapping = $autoMapping;
+        $this->validator = $validator;
+        $this->adminStoreOwnerPaymentService = $adminStoreOwnerPaymentService;
+    }
+
+    /**
+     * admin:Create new Payment from store
+     * @Route("storeownerpayment", name="createStoreOwnerPayment", methods={"POST"})
+     * @IsGranted("ROLE_ADMIN")
+     * @param Request $request
+     * @return JsonResponse
+     *
+     * @OA\Tag(name="Store Owner Payment")
+     *
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="new payment",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="amount"),
+     *          @OA\Property(type="number", property="store"),
+     *      )
+     * )
+     *
+     * @OA\Response(
+     *      response=201,
+     *      description="Returns new payment",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
+     *            @OA\Property(type="integer", property="id"),
+     *            @OA\Property(type="string", property="amount"),
+     *            @OA\Property(type="string", property="date"),
+     *      )
+     *   )
+     * )
+     *
+     * @Security(name="Bearer")
+     */
+    public function createStoreOwnerPayment(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, AdminStoreOwnerPaymentCreateRequest::class, (object)$data);
+
+        $violations = $this->validator->validate($request);
+
+        if (\count($violations) > 0) {
+            $violationsString = (string) $violations;
+
+            return new JsonResponse($violationsString, Response::HTTP_OK);
+        }
+
+        $result = $this->adminStoreOwnerPaymentService->createStoreOwnerPayment($request);
+
+        return $this->response($result, self::CREATE);
+    }
+
+    /**
+     * admin:Create new Payment from store
```

Comment:
I guess should name the api: update payment, not create.

Reply:
why?
The admin is the one who adds the payment not update